source/extensions/filters/http/lua/lua_filter.cc

```diff
@@ -23,7 +23,10 @@ StreamHandleWrapper::StreamHandleWrapper(Filters::Common::Lua::Coroutine& corout
         if (state_ == State::Running) {
           throw Filters::Common::Lua::LuaException("script performed an unexpected yield");
         }
-      }) {}
+      }),
+      fireAndForgetHttpWriter_(new FireAndForgetHttpWriter(filter)) {}
```

Comment:
no reason to heap allocate this, just use it as a non-pointer value

```FireAndForgetHttpWriter fire_and_forget_http_writer_;```

and just `fire_and_forget_http_writer_(filter)` in the constructor

Reply:
This actually does need to be heap allocated because the success/failure callbacks can live beyond the filter, but not in the way it is done here. The Lua filter config needs to have a fire and forget writer that works identically to the router's shadow writer. Take a look at the pattern that is used there.

Reply:
(Or yes per other comment just pass static callbacks and that should be fine also).